src/libra_py/workflows/nbra/lz.py

```diff
@@ -228,6 +228,117 @@ def Belyaev_Lebedev(Hvib, params):
 
 
 
+
+def adjust_SD_probabilities(P, params):
+    """
+    Adjusts the surface hopping probabilties computed according to Belyaev-Lebedev's work by setting the probability
+    to hop between Slater determinants that differ by more than 1 electron to zero.  
+
+    Args:
+        P ( list of lists of MATRIX ): each element of P contains the probability matrix according to the NBRA BLSH method
+        params ( dictionary ): control parameters
+
+        * **params["excitations"]** ( list of lists of lists ) : SDs themselves (QE or dftb+ non-tddftb) or the SD transitions (tddftb). If the excitations are non-changing, then the
+                                                                 length of excitations is 1. If excitations do change, then there must be an excitation list for each step. In this case,
+                                                                 the length of excitation = nsteps = len(P)     
+            Examples:
+                
+                Assume we have constructed SDs from a basis of 4 alpha and 4 beta Kohn-Sham spin-orbitals
+                The ground state is defined as: [1, 2, -5, -6]. Consider only alpha-excitations
+
+                We currently do non-tddftb computations by forming our own SDs in libra_py.workflows.nbra.step3
+
+                - QE or dftb+ non-tddftb - [ [ [1, 3, -5, -6], [1, 4, -5, -6], [3, 2, -5, -6], [4, 2, -5, -6] ] ] 
+
+                As tddftb is used exclusively with the BLSH-NBRA method, we usually just read the energies, and know information only regarding
+                the index of the orbtial transitions, such as: [2 -> 3]. So, we package this in the following form
+
+                - tddftb - [ [ [2,3], [2,4], [1,3], [1,4] ] ]                                           
+    """
+
+    if "excitations" not in params:

```

Comment:
this is a wrong verification - you should be looking if "excitation" is in the keys of the params dictionary, not as an entry of it